modules/core/src/main/java/org/apache/ignite/internal/util/GridUnsafe.java

```diff
@@ -1361,6 +1361,42 @@ public static boolean compareAndSwapLong(Object obj, long off, long exp, long up
         return UNSAFE.compareAndSwapLong(obj, off, exp, upd);
     }
 
+    /**
+     * Atomically increments value stored in an integer pointed by {@code ptr}.
+     *
+     * @param ptr Pointer to an integer.
+     * @return Updated value.
+     */
+    public static int incrementAndGetInt(long ptr) {
+        while (true) {
```

Comment:
Method body could be replaced by `Unsafe.getAndAddInt(null, ptr, 1) + 1`.

Reply:
Done.